chore: Update commit validation to use commitlint

Patil2099 opened the pull request:
Fixes #62

lychyi commented on package.json:
```diff
@@ -130,5 +130,8 @@
   ],
-  "browserslist": "> 5%, last 2 firefox versions, last 2 chrome versions, last 2 safari versions, last 2 edge versions, ie 11"
+  "browserslist": "> 5%, last 2 firefox versions, last 2 chrome versions, last 2 safari versions, last 2 edge versions, ie 11",
+  "dependencies": {
```
This changeset is odd, did you add `commitlint` to dependencies and then remove?

NicholasBoll replied:
Agreed. I can fix this and merge today

Patil2099 commented:
I added commitlint to devdependencies and removed it from there

On Aug 20, 2019 3:29 AM, Lynn Chyi <notifications@github.com> wrote:

@lychyi commented on this pull request.

________________________________

In package.json<https://github.com/Workday/canvas-kit/pull/124#discussion_r315430457>:

> @@ -130,5 +130,8 @@
   "workspaces": [
     "modules/**"
   ],
-  "browserslist": "> 5%, last 2 firefox versions, last 2 chrome versions, last 2 safari versions, last 2 edge versions, ie 11"
+  "browserslist": "> 5%, last 2 firefox versions, last 2 chrome versions, last 2 safari versions, last 2 edge versions, ie 11",
+  "dependencies": {


This changeset is odd, did you add commitlint to dependencies and then remove?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/Workday/canvas-kit/pull/124?email_source=notifications&email_token=AIQAR5E26LY4JTL3ZQONCSLQFMJS5A5CNFSM4IL3KKD2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCCADLOY#pullrequestreview-276837819>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AIQAR5HY6VCETSHNMGQBZ3LQFMJS5ANCNFSM4IL3KKDQ>.

NicholasBoll commented:
This PR isn't complete. When creating a commit, the command `npm run -s commitmsg` is run first which right now runs `validate-commit-msg` which no longer exists. The `commitmsg` command in `package.json` needs to be updated to handle linting correctly.

NicholasBoll commented:
I'm following https://github.com/conventional-changelog/commitlint#getting-started to see how `commitlint` works

NicholasBoll commented:
I finished this PR. I'll let the team decide on the commit message format. `commitlint` is more strict than what we had before. It follows https://www.conventionalcommits.org/en/v1.0.0-beta.4/:

Before:
```
# No scope checks, can have sentence-case
fix(WhatEverYouWant): Fix something here
```

After
```
# Scope is checked against `modules` folder, cannot be sentence-case
fix(button): remove extra padding
```

Not a breaking change.

### Suggested squash message:
chore: update commit validation to use commitlint

**Note**
Squash commit messages are not checked!
